sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,414 @@ def to_rotation_matrix(self, v=None):
+
+        M = Matrix([[q1.b, q1.c, q1.d], [q2.b, q2.c, q2.d], [q3.b, q3.c, q3.d]]).det()
+        return M.is_zero
+
+    def parallel(self, other):
+        """
+        Returns True if the two pure quaternions seen as 3D vectors are parallel else returns False.
+
+        Explanation
+        ===========
+
+        Two pure quaternions are called parallel when their product is commutative.
+
+        Parameters
+        ==========
+
+        other : a Quaternion
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(0, 4, 4, 4)
+        >>> q1 = Quaternion(0, 8, 8, 8)
+        >>> q.parallel(q1)
+        True
+
+        >>> q1 = Quaternion(0, 8, 13, 12)
+        >>> q.parallel(q1)
+        False
+
+        """
+
+        if fuzzy_not(self.is_pure()) or fuzzy_not(other.is_pure()):
+            raise ValueError('The provided quaternions must be pure')
+
+        return (self*other - other*self).is_zero_quaternion()
+
+    def orthogonal(self, other):
+        """
+        Returns True if the two pure quaternions seen as 3D vectors are orthogonal else returns False.
+
+        Explanation
+        ===========
+
+        Two pure quaternions are called orthogonal when their product is anti-commutative.
+
+        Parameters
+        ==========
+
+        other : a Quaternion
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(0, 4, 4, 4)
+        >>> q1 = Quaternion(0, 8, 8, 8)
+        >>> q.orthogonal(q1)
+        False
+
+        >>> q1 = Quaternion(0, 2, 2, 0)
+        >>> q = Quaternion(0, 2, -2, 0)
+        >>> q.orthogonal(q1)
+        True
+
+        """
+
+        if fuzzy_not(self.is_pure()) or fuzzy_not(other.is_pure()):
+            raise ValueError('The provided quaternions must be pure')
+
+        return (self*other + other*self).is_zero_quaternion()
+
+    def index_vector(self):
+        """
+        Returns the index vector of the quaternion.
+
+        Explanation
+        ===========
+
+        Index vector is given by mod(q) Ax(q) where Ax(q) is the axis of the quaternion q.
+
+        Returns
+        =======
+
+        Quaternion: representing index vector of the provided quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(2, 4, 2, 4)
+        >>> q.index_vector()
+        0 + 4*sqrt(10)/3*i + 2*sqrt(10)/3*j + 4*sqrt(10)/3*k
+
+        >>> q = Quaternion(0, 2, 2, 0)
+        >>> q.index_vector()
+        0 + 2*i + 2*j + 0*k
+
+        See Also
+        ========
+
+        axis
+
+        """
+
+        return self.norm() * self.vector_part().normalize()
+
+    def mensor(self):
+        """
+        Returns the mensor of the quaternion.
+
+        Returns
+        =======
+
+        expr: real number represnting the mensor of the quaternion.
```

Comment:
You should define the mensor (it's the log of the norm) and then maybe have a see also for e.g. norm.